### ISSUE TITLE
REPL does not launch with M-x cider-jack-in for a Clojure file in a simple Boot project

### ISSUE THREAD

**Issue by gsuzor91** [clojure-emacs/cider#2893]
Hello, with cider 0.26.1 I am not able to launch a REPL with cider-jack-in or the equivalent key chord for a very simple Boot project.

I fortunately had a copy of cider 0.25.0 on another computer. Rolling back to that resolves my issue. Upgrading again to 0.26.1 the issue returns. Digging a little deeper in cider 0.26.1, I found that I can change the nrepl cider-jack-in-dependencies from "0.8.0" to "0.7.0" on line 385 in cider.el. This also resolves my issue, but realize this is more of a workaround instead of fixing the underlying issue.

## Expected behavior
REPL should launch when calling M-x cider-jack-in on a Clojure file in a simple Boot project.

## Actual behavior
REPL does not launch. See below for errors.

#### minibuffer
````
error in process filter: Sync nREPL request timed out (op clone id 1)
````

#### \*Messages\*
````
[nREPL] Starting server via "c:/dev-programs/boot-clj/boot.exe" -i "(require 'cider.tasks)" -d "nrepl:0.8.0" -d "cider/cider-nrepl:0.25.3" cider.tasks/add-middleware -m "cider.nrepl/cider-middleware" repl -s -b localhost wait
[nREPL] server started on 58746
[nREPL] Establishing direct connection to localhost:58746 ...
[nREPL] Direct connection to localhost:58746 established
error in process filter: nrepl-send-sync-request: Sync nREPL request timed out (op clone id 1)
error in process filter: Sync nREPL request timed out (op clone id 1)
````
#### \*nrepl-server clojure/dice:localhost\*
````
[WARNING] No nREPL middleware descriptor in metadata of null, see nrepl.middleware/set-descriptor!
nREPL server started on port 58746 on host localhost - nrepl://localhost:58746
ERROR: Unhandled REPL handler exception processing message {:op clone, :id 1}
java.lang.NullPointerException
	at nrepl.middleware.dynamic_loader$wrap_dynamic_loader$fn__4732.invoke(dynamic_loader.clj:97)
	at nrepl.middleware$wrap_conj_descriptor$fn__4231.invoke(middleware.clj:16)
	at nrepl.server$default_handler$fn__5158.invoke(server.clj:130)
	at nrepl.server$handle_STAR_.invokeStatic(server.clj:22)
	at nrepl.server$handle_STAR_.invoke(server.clj:19)
	at nrepl.server$handle$fn__5129.invoke(server.clj:39)
	at clojure.core$binding_conveyor_fn$fn__5754.invoke(core.clj:2030)
	at clojure.lang.AFn.call(AFn.java:18)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:823)
````

## Steps to reproduce the problem
Below are copies of the files used that replicate the issue on my system.
Running M-x cider-jack-in with focus on dice/src/dice/core.clj in Emacs will recreate the issue.

#### ~/.boot/boot.properties
````
#http://boot-clj.com
#Sat Aug 22 20:59:35 EDT 2020
BOOT_CLOJURE_NAME=org.clojure/clojure
BOOT_CLOJURE_VERSION=1.10.1
BOOT_VERSION=2.8.3
````

#### dice/build.boot
````
(set-env!
 :resource-paths #{"src"})

(task-options!
 pom {:project 'dice
      :version "0.1.0"})
````

#### dice/src/dice/core.clj
```
(ns dice.core)

(defn roll [n]
  (take n (repeatedly #(+ (rand-int 6) 1))))
```

### CIDER version information
CIDER 0.26.1

### Boot version
Boot 2.8.3

### Emacs version
GNU Emacs 26.3 (build 1, x86_64-w64-mingw32) of 2019-08-29

### Java version
AdoptOpenJDK
jdk-8.0.265.01-openj9

### Operating system
Windows 10 Pro 2004 (19041.450)

**Comment by bbatsov** [comment]
I don't think that's a bug with CIDER itself, but rather with nREPL 0.8 that it uses by default when jacking-in. @shen-tian can you check that nREPL 0.8 works properly with Boot?

**Comment by shen-tian** [comment]
Just had a go at reproduction, and `cider-jack-in` succeeded:

```
;; Connected to nREPL server - nrepl://localhost:49635
;; CIDER 0.26.1 (Nesebar), nREPL 0.8.0
;; Clojure 1.10.1, Java 14.0.1
;;     Docs: (doc function-name)
;;           (find-doc part-of-name)
;;   Source: (source function-name)
;;  Javadoc: (javadoc java-object-or-class)
;;     Exit: <C-c C-q>
;;  Results: Stored in vars *1, *2, *3, an exception in *e;
;;  Startup: /usr/local/bin/boot -i "(require 'cider.tasks)" -d nrepl\:0.8.0 -d refactor-nrepl\:2.5.0 -d cider/cider-nrepl\:0.25.3 cider.tasks/add-middleware -m refactor-nrepl.middleware/wrap-refactor -m cider.nrepl/cider-middleware repl -s -b localhost wait
boot.user> 
```

The odd thing is that I'm somehow using Java 14, which is a boot-clj dependency. Any ideas about that? (I usually use jenv to manage java versions, and get this in a terminal in the `dice` project:

```
Shens-Mac-mini:~ shen$ java -version
openjdk version "1.8.0_202"
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_202-b08)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.202-b08, mixed mode)
```

**Comment by bbatsov** [comment]
I guess you can check Emacs's `exec-path` and see what's on it.

**Comment by shen-tian** [comment]
I also see my `jack-in` added refactor. What influences that?

**Comment by bbatsov** [comment]
Whether `clj-refactor.el` is installed or not.

**Comment by ahwatts** [comment]
As noted above, I filed a ticket with nREPL, as I can reproduce this without CIDER.

I'm seeing this issue on Windows, and only on Windows, and have an even more minimal reproduction:

```
C:\Users\Andrew Watts> boot -d nrepl:0.8.0 repl
[WARNING] No nREPL middleware descriptor in metadata of null, see nrepl.middleware/set-descriptor!
nREPL server started on port 50607 on host 127.0.0.1 - nrepl://127.0.0.1:50607
ERROR: Unhandled REPL handler exception processing message {:id 2995ad98-b1e6-479c-af71-244ea844862c, :op clone}
java.lang.NullPointerException
        at nrepl.middleware.dynamic_loader$wrap_dynamic_loader$fn__4663.invoke(dynamic_loader.clj:97)
        at nrepl.middleware$wrap_conj_descriptor$fn__4162.invoke(middleware.clj:16)
        at nrepl.server$default_handler$fn__5089.invoke(server.clj:130)
        at nrepl.server$handle_STAR_.invokeStatic(server.clj:22)
        at nrepl.server$handle_STAR_.invoke(server.clj:19)
        at nrepl.server$handle$fn__5060.invoke(server.clj:39)
        at clojure.core$binding_conveyor_fn$fn__5754.invoke(core.clj:2030)
        at clojure.lang.AFn.call(AFn.java:18)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
```

If I switch the `nrepl` dependency to 0.7.0, it works fine.

**Comment by bbatsov** [comment]
This was fixed in nREPL 0.8.1 (released yesterday).

**Comment by ahwatts** [comment]
A workaround is to set `cider-jack-in-dependencies` to use nREPL 0.7.0, which I do with this:

```elisp
(setq cider-jack-in-dependencies
      (cons '("nrepl" "0.7.0")
            (seq-filter (lambda (e)
                          (not (equal "nrepl" (car e))))
                        cider-jack-in-dependencies)))
```

(That snippet is probably overkill given that `cider-jack-in-dependencies` only includes nREPL by default...)
